### PR TITLE
Update sha2, ed25519 and curve25519-dalek dependencies

### PR DESCRIPTION
**Pull request by remoun**
Updates `sha2`, `ed25519`, `curve25519-dalek` dependencies to latest crates.
Temporarily patched `curve25519-dalek` pending https://github.com/mobilecoinfoundation/curve25519-dalek/pull/2

Also fixes a warning in `tests/ed25519`.

Supports https://github.com/mobilecoinfoundation/mobilecoin/issues/1326

### CODE REVIEW

**Review by cbeck88 — DISMISSED** [review]
LGTM except for personal fork

**Review by jcape — DISMISSED** [review]
LGTM pending curve25519-dalek location update.

**Review by cbeck88 — DISMISSED** [review]

**Comment by remoun** [comment]
PTAL, I've updated the patched curve dep to use the MCF fork.

**Review by jcape — DISMISSED** [review]

**Review by integral-llc — DISMISSED** [review]

**Comment by remoun** [comment]
PTAL, I had to rebase the commits to sign them.

File: Cargo.toml [hunk 1]
```diff
@@ -63,3 +63,6 @@ legacy_compatibility = []
 u64_backend = ["curve25519-dalek/u64_backend"]
 u32_backend = ["curve25519-dalek/u32_backend"]
 simd_backend = ["curve25519-dalek/simd_backend"]
+
+[patch.crates-io]
+curve25519-dalek = { git = "https://github.com/mobilecoinfoundation/curve25519-dalek.git", rev = "8791722e0273762552c9a056eaccb7df6baf44d7" }
```
Comment (remoun) [inline Cargo.toml:68]: This can be removed when https://github.com/dalek-cryptography/curve25519-dalek/pull/383 is published

**Review by jcape — APPROVED** [review]

**Pull request merged**